### PR TITLE
docs(Spanner): Updated Connection String documentation.

### PR DESCRIPTION
**Pull request by Rishabh-V**
Please let me know if any changes are required and I will try to do them quickly.

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 1]
```diff
@@ -134,3 +134,54 @@ documentation.
...
+Examples:
+
+- ClrToSpannerTypeDefaultMappings=DecimalToFloat64,DateTimeToDate
+- ClrToSpannerTypeDefaultMappings=DecimalToNumeric,DateTimeToTimestamp
+
+## SpannerToClrTypeDefaultMappings
+
+Option to configure the default `SpannerDbType` to CLR type mappings. This option comes into picture 
+only if CLR type of the value being read is not explicitly provided while reading the data from the database.
+E.g., `ExecuteScalarAsync<object>()`. Currently only `SpannerDbType.Date` is supported.
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/docs/connection_string.md:174]: I'd put the example at the end of the previous sentence, and I'd probably use `reader["fieldName"]` as the example. (I think it's somewhat unlikely for people to *explicitly* specify a type argument of `object` - it's more likely that they'll use non-generic APIs such as the reader indexer.)

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 2]
```diff
@@ -134,3 +134,54 @@ documentation.
...
+
+The valid type mappings for `SpannerDbType.Date` are:
+
+- DateToDateTime - `SpannerDbType.Date` will map to `DateTime` if CLR type of the value being read is not explicitly provided.
+- DateToSpannerDate - `SpannerDbType.Date` will map to `SpannerDate` if CLR type of the value being read is not explicitly provided.
+
+The mapping can be provided as comma separated values. Only one mapping for a `SpannerDbType` must be provided.
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/docs/connection_string.md:181]: Ditto earlier comments

**Review by jskeet — COMMENTED** [review]
Mostly fine - nits for readability really.

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 3]
```diff
@@ -134,3 +134,54 @@ documentation.
...
+
+## ClrToSpannerTypeDefaultMappings
+
+ Option to configure the default CLR type to `SpannerDbType` mapping. 
+ This option comes into picture only if `SpannerDbType` and `DbType` of the `SpannerParameter`
+ are not explicitly provided. Currently only `decimal` and `DateTime` CLR types are supported.
+
+ The valid type mappings for `decimal` are:
+
+- DecimalToFloat64 - `decimal` CLR type will map to `SpannerDbType.Float64` if `SpannerDbType`
```
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply apis/Google.Cloud.Spanner.Data/docs/connection_string.md]: Done.

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 4]
```diff
@@ -134,3 +134,54 @@ documentation.
...
+- DateTimeToTimestamp - `DateTime` CLR type will map to `SpannerDbType.Timestamp` if `SpannerDbType` 
+and `DbType` are not explicitly provided for the `SpannerParameter`.
+
+The mapping can be provided as comma separated values. Only one mapping for a type must be provided. 
+Providing mutiple mapping for a type, or providing invalid mapping or providing whitespaces will result in `ArgumentException`.
+
+Examples:
+
+- ClrToSpannerTypeDefaultMappings=DecimalToFloat64,DateTimeToDate
+- ClrToSpannerTypeDefaultMappings=DecimalToNumeric,DateTimeToTimestamp
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/docs/connection_string.md:162]: Ping on this one?

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 5]
```diff
@@ -134,3 +134,47 @@ documentation.
...
+
+Examples:
+
+- ClrToSpannerTypeDefaultMappings=DecimalToFloat64
+- ClrToSpannerTypeDefaultMappings=DecimalToNumeric,DateTimeToTimestamp
+
+## SpannerToClrTypeDefaultMappings
+
+Option to configure the default `SpannerDbType` to CLR type mappings. This option is only used if the 
+CLR type of the value being read is not explicitly provided while reading the data from the database. E.g.,
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/docs/connection_string.md:167]: All I was suggesting in terms of location was changing this to:

```text
[...] while reading the data from the database, for example using an indexer: `reader["fieldName"]`.
```

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md [hunk 6]
```diff
@@ -134,3 +134,47 @@ documentation.
...
+Option to configure the default `SpannerDbType` to CLR type mappings. This option is only used if the 
+CLR type of the value being read is not explicitly provided while reading the data from the database. E.g., 
+`reader["fieldName"]`. Currently only `SpannerDbType.Date` is supported.
+
+The valid type mappings for `SpannerDbType.Date` are:
+
+- DateToDateTime - `SpannerDbType.Date` will map to `DateTime`.
+- DateToSpannerDate - `SpannerDbType.Date` will map to `SpannerDate`.
+
+The mapping can be provided as comma separated values. For each `SpannerDbType`, at most one mapping may be provided.
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/docs/connection_string.md:175]: It's just struck me that until we've got multiple types to map, this doesn't make much sense - it's impossible to provide more than one value and still have a valid string. I think it's fine to leave though.

**Review by Rishabh-V — COMMENTED** [review]
Done. One mapping example started reflecting after a while, but I see it now.

File: apis/Google.Cloud.Spanner.Data/docs/connection_string.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/docs/connection_string.md:162]: Ah, sorry, I didn't spot that you'd changed the first one.

**Review by jskeet — APPROVED** [review]

**Pull request merged**